File: java/Decorator_Pattern/decorator/Mocha.java
package java.Decorator_Pattern.decorator;

import beverage.Beverage;

public class Mocha extends CondimentDecorator {
    
    public Mocha(Beverage beverage){
        this.beverage = beverage;
    }

    @Override
    public double cost(){
        return beverage.cost() + .20;
    }

    @Override
    public String getDescription(){
        return beverage.getDescription() + ", Mocha";
    }
}

File: problems/coffee-machine/mocha.py
from condiment_decorator import CondimentDecorator

class Mocha(CondimentDecorator):
    def get_description(self) -> str:
        return self.beverage.get_description() + ", Mocha"

    def get_cost(self) -> float:
        return self.beverage.get_cost() + .50